core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java

```diff
@@ -101,15 +104,18 @@ public void start(String channelId) throws UnauthenticatedException, Unavailable
       throw new UnauthenticatedException(se.getMessage(), se);
     } catch (InterruptedException ie) {
       Thread.currentThread().interrupt();
-      throw new UnauthenticatedException(ie.getMessage(), ie);
+      throw new UnavailableException(ie.getMessage(), ie);
     } catch (ExecutionException e) {
-      Throwable cause = e.getCause();
+      Throwable cause = (e.getCause() != null) ? e.getCause() : e;
       if (cause != null && cause instanceof StatusRuntimeException) {
-        if (((StatusRuntimeException) cause).getStatus().getCode() == Status.Code.UNAVAILABLE) {
-          throw new UnavailableException(cause.getMessage(), cause);
+        StatusRuntimeException sre = (StatusRuntimeException) cause;
+        // If caught unimplemented, that means server does not support authentication.
```

Comment:
As long as we can confirm this is always true I am okay with blanketing the `UnimplementedException` to `UnauthenticatedException`.

Reply:
There is no other call made from here.